### ISSUE TITLE
[1.8.0] displayed Transfer speed Higher than limit

### ISSUE THREAD

**Issue by cdamken** [owncloud/client#3095]
### Expected behavior
Limit upload speed to 50 KB/s
Upload to 50KB/s

### Actual behavior
Limit upload speed to 50 KB/s
Upload to 70KB/s
![speed](https://cloud.githubusercontent.com/assets/7264495/7089065/5eb2a936-df99-11e4-972b-ee48d9f47cde.png)

### How to reproduce it:
Sync-client 1.8.0 
1.- Add a big File to the upload queue (I added a 2 GB file)
2.- Change the transfer sped to 50 KB/s
3.- Pause 
4.- Resume
5.- Wait until the speed stabilizes.

Tested with different speeds.
when 5KB/s
upload 8KB/s

@MorrisJobke 

00002794

**Comment by dragotin** [comment]
@ogoffart mind to check this? Maybe it's just fuzzy calculation in the progress widget.

**Comment by ogoffart** [comment]
What platform ?  (which version of Qt)

CC @guruz

**Comment by guruz** [comment]
Carlos, have you tried a system tool that displays actual speed? I dont know if windows task manager can do it

**Comment by cdamken** [comment]
@ogoffart  Windows  with 1.8.0. How can I find out the Qt version?

@guruz I will test wit another tool, it's hard to see directly in the Windows network notification, but anyway the widget tells another speed.

**Comment by caratsch** [comment]
@guruz the speed looks right on the system (Resource Monitor shows the right speed). The sync client shows now 260 KiB/s, but only 10 KiB/s are allowed and currently transferred.

**Comment by cdamken** [comment]
@guruz Then the only problem is the speed shown in the widget

**Comment by ogoffart** [comment]
We probably should remove that speed there, it is totaly not accurate.

**Comment by guruz** [comment]
...let's make it a rolling average calculation instead of momentary speed display.

**Comment by ckamm** [comment]
@guruz I've code for that ready to be merged, pending review. It changes it to an exponential moving average. See #2328. Maybe I'll just merge these to master for testing...

**Comment by guruz** [comment]
@ckamm Sounds good.. :)

**Comment by cdamken** [comment]
Is something fixed in 1.8.1? Should I test it?

**Comment by ckamm** [comment]
@cdamken Not that I'm aware. There are some patches in master (for 2.0).

**Comment by guruz** [comment]
@cdamken Can you please try ownCloud-2.0.0.5092-nightly20150520-setup.exe ?

**Comment by cdamken** [comment]
@guruz Where can I have it? http://download.owncloud.com/desktop/daily/ here is not available.

**Comment by danimo** [comment]
@cdamken build 5109 should show up in that location in a few minutes.

**Comment by caratsch** [comment]
@guruz Seems not to be fixed in the build 5110 nightly. At the beginning it starts at about 60 MB/s and goes slowly down until 3.3 KB/s, but it's uploading with 10KB/s the whole time.
![capture](https://cloud.githubusercontent.com/assets/8424737/7854216/97063462-0517-11e5-91d0-3bfb34b32b3e.PNG)

**Comment by cdamken** [comment]
@danimo I just tested the 5110 too, same results like @caratsch

**Comment by MTRichards** [comment]
So this will be resolved with the release of 1.8.3?

**Comment by dragotin** [comment]
Probably not, I am not aware that we pushed something to 1.8 branch. @ckamm anything we can/want to backport?

**Comment by ckamm** [comment]
@dragotin I don't think this is something critical or safe enough to backport to 1.8 at this stage.

**Comment by cdamken** [comment]
@ckamm @danimo IMHO no need to backport, but in 2.0 would nice to have it

**Comment by ogoffart** [comment]
Remember that the thing only show one of the files and there are several items being downloaded in parallel.

**Comment by cdamken** [comment]
@ogoffart I can't remember something that nobody told me :smile: ... The same thing will happen with the end user that the just see in the list 1 file in the syn-client and the other 2 are hidden.
(It makes a similar confusion with MB and MiB) 

How about displaying the 3 files that are being uploaded or downloaded?

**Comment by guruz** [comment]
> How about displaying the 3 files that are being uploaded or downloaded?

Will be in next nightly :-) Please test then..

**Comment by guruz** [comment]
@cdamken  http://dl-oc.owncloud.com/desktop/daily/ownCloud-2.0.0.5361-nightly20150806-setup.exe

**Comment by phil-davis** [comment]
2.0.0.5361 on Windows - working and showing me the 3 files and a total download speed that is very believable for my link.
After the 3 filenames that are comma-separated there is a dash (or underscore?) followed by the speed (e.g. the 69 kB/s in the screen shot). That "-" has a weird gray-shade vertical bar rendered also.
I wonder where that artifact comes from?

**Comment by cdamken** [comment]
@guruz I need to uninstall the previous version to install the new one?

![error](https://cloud.githubusercontent.com/assets/7264495/9115271/7e390102-3c5e-11e5-8774-739d2b81ed9f.png)

**Comment by danimo** [comment]
@cdamken That was a one-time glitch. Simply install a new version of the testpilot cloud, and choose not to uninstall the old one after the upgrade.

**Comment by cdamken** [comment]
@danimo Ok

**Comment by guruz** [comment]
@phil-davis Hmpf, on my OS X those are nice UTF-8 arrows. Why does this not work :-|
https://github.com/owncloud/client/blob/master/src/gui/folderstatusmodel.cpp#L712

Thanks for testing

**Comment by cdamken** [comment]
![speed](https://cloud.githubusercontent.com/assets/7264495/9116209/72bf6d66-3c63-11e5-88c5-2193aa71434b.png)

**Comment by cdamken** [comment]
@guruz I press enter :smile:  Is there a 1 before the 3.7kB/s??

**Comment by phil-davis** [comment]
Before the "3.7MB/s" it is supposed to be a down-arrow.
I tried in MS Word on this same computer doing Alt-8595 (2193 hex = 8595 decimal) and it displays a nice down-arrow in Calibri font. So Microsoft does at least know how to render a the down-arrow in at least 1 font :)

**Comment by cdamken** [comment]
@phil-davis Yes, that should be an arrow, probably my display does not show it right, it's in a VM

**Comment by guruz** [comment]
^^ @jturcotte Maybe you know if this is a Qt issue? @danimo claims it is a font issue.

In any case, i'll ifdef something for Windows for now

**Comment by phil-davis** [comment]
This gives an easy way to verify the new "feature" that downloads are serialized when a download bandwidth is specified in the Network settings. I set Download Bandwidth to 20KBytes/s and added a few files directly on the server. The progress display shows them downloading 1 at a time at around 20kB/s. The same works for uploading.
Note that if there is either a download or upload speed limit set, then file transfers are serialized in both directions. e.g. If I set a download bandwidth, and then add a dozen files to the client to upload, then the upload is done 1 file at a time. I think that is probably reasonable.

**Comment by guruz** [comment]
@cdamken @phil-davis beta1 should not use the UTF-8 arrows on Windows anymore.

**Comment by phil-davis** [comment]
Confirmed good on 2.0.0beta1 on Windows. The words "upload" and "download" appear in the text rather than the arrows.

**Comment by cdamken** [comment]
:+1:

**Comment by phil-davis** [comment]
The "arrows" rendering on Windows has been fixed now by changing it to the words "upload" and "download". 2.0.0beta1 and later.
